libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/json/MapJsonReader.kt

```diff
@@ -423,3 +411,22 @@ constructor(
     }
   }
 }
+
+/**
+ * A typealias for a type-unsafe Kotlin representation of JSON. This typealias is
+ * mainly for internal documentation purposes and low-level manipulations and should
+ * generally be avoided in application code.
+ *
+ * [ApolloJsonElement] can be any of:
+ * - String
+ * - Int
+ * - Double
+ * - Long
+ * - JsonNumber
+ * - null
+ * - Map<String, ApolloJsonElement> where values are any of these values recursively
+ * - List<ApolloJsonElement> where values are any of these values recursively
+ *
+ * Anything else is undefined
+ */
+typealias ApolloJsonElement = Any?
```

Comment:
Interesting that this doesn't show in apiDump. I guess it's source-breaking only. 

In all cases, I wish I've done that earlier to be able to point in KDoc to what `Any?` can contain. Naming is very open to suggestions.

Reply:
`ApolloJsonElement` sounds good to me 👍. I remember doing a similar thing (`typealias JsonMap = Map<String, Any?>`) in `DeferredJsonMerger` which made the code less messy.